### ISSUE TITLE
Suggest a Grace Time based on average job execution time

### ISSUE THREAD

**Issue by danemacmillan** [healthchecks/healthchecks#218]
I've begun [measuring job execution time](https://healthchecks.io/docs/#start-event) of all of the health checks that were deployed before the feature was available. One thing that occurred to me is that, given a large enough dataset of measured execution times, the Grace Time feature could be more empowered based on the data being collected, and help users set more informed values.

I don't think I'd want the Grace Time to be automatically modified based on this data, in the off chance that some jobs are severely skewed, but I think providing the average execution time for a job somewhere in the interface, and offering it up as a suggested good value based on `n` jobs, would be really insightful. I would even say this data should only be made available to users who have collected a certain threshold of data--say, 100 successfully measured job runs per job--in order to appear in the UI somewhere.

The reason I bring this up is because I'm just manually setting Grace Times based on my anecdotal knowledge of their average run times; this is usually sufficient. Most of us using this service are likely intimately aware of their system operations. However, there are some jobs that take longer due to the simple fact that these jobs have stayed the same but the amount of data they process has increased. I'm not obsessing about how long these particular jobs run and then tweaking their grace times accordingly, but typically the moment I decide to increase a Grace Time is, for example, after a four month period I note that I'm now getting too many alerts for a particular job, and it's simply because it requires more time to process more data. In this example I would check how long it now takes, and then adjust the Grace Time accordingly.

To summarize, then, I think displaying the average execution time for each job somewhere in the UI would help users in setting more informed Grace Time values.

**Comment by skorokithakis** [comment]
That's a pretty cool feature, would you accept a PR if I had time to work on it?

**Comment by cuu508** [comment]
Yes, I think this would be a neat and useful feature. I'm open to reviewing and accepting PR(s).
